File: GNSDK/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/GnExternalId.java

package com.gracenote.gnsdk;

/** 
* Third-party identifier used to match identified media to merchandise IDs in online stores and other services 
* <p> 
*/ 
 
public class GnExternalId extends GnDataObject {
  private long swigCPtr;

  protected GnExternalId(long cPtr, boolean cMemoryOwn) {
    super(gnsdk_javaJNI.GnExternalId_SWIGUpcast(cPtr), cMemoryOwn);
    swigCPtr = cPtr;
  }

  protected static long getCPtr(GnExternalId obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_GnExternalId(swigCPtr);
      }
      swigCPtr = 0;
    }
    super.delete();
  }

  public static String gnType() {
    return gnsdk_javaJNI.GnExternalId_gnType();
  }

  public static GnExternalId from(GnDataObject obj) throws com.gracenote.gnsdk.GnException {
    return new GnExternalId(gnsdk_javaJNI.GnExternalId_from(GnDataObject.getCPtr(obj), obj), true);
  }

/** 
*  External ID source (e.g., Amazon) 
*  @return External ID source 
*/ 
 
  public String source() {
    return gnsdk_javaJNI.GnExternalId_source(swigCPtr, this);
  }

/** 
*  External ID type 
*  @return External ID type 
*/ 
 
  public String type() {
    return gnsdk_javaJNI.GnExternalId_type(swigCPtr, this);
  }

/** 
*  External ID value 
*  @return External ID value 
*/ 
 
  public String value() {
    return gnsdk_javaJNI.GnExternalId_value(swigCPtr, this);
  }

}

File: GNSDK/wrappers/gnsdk_csharp/src_wrapper/GnExternalId.cs

namespace GracenoteSDK {

using System;
using System.Runtime.InteropServices;

/**
* Third-party identifier used to match identified media to merchandise IDs in online stores and other services
*
*/
public class GnExternalId : GnDataObject {
  private HandleRef swigCPtr;

  internal GnExternalId(IntPtr cPtr, bool cMemoryOwn) : base(gnsdk_csharp_marshalPINVOKE.GnExternalId_SWIGUpcast(cPtr), cMemoryOwn) {
    swigCPtr = new HandleRef(this, cPtr);
  }

  internal static HandleRef getCPtr(GnExternalId obj) {
    return (obj == null) ? new HandleRef(null, IntPtr.Zero) : obj.swigCPtr;
  }

  ~GnExternalId() {
    Dispose();
  }

  public override void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_GnExternalId(swigCPtr);
        }
        swigCPtr = new HandleRef(null, IntPtr.Zero);
      }
      GC.SuppressFinalize(this);
      base.Dispose();
    }
  }

  public static string GnType() {
    string ret = gnsdk_csharp_marshalPINVOKE.GnExternalId_GnType();
    return ret;
  }

  public static GnExternalId From(GnDataObject obj) {
    GnExternalId ret = new GnExternalId(gnsdk_csharp_marshalPINVOKE.GnExternalId_From(GnDataObject.getCPtr(obj)), true);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
    return ret;
  }

/**
*  External ID source (e.g., Amazon)
*  @return External ID source
*/
  public string Source {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnExternalId_Source_get(swigCPtr) );
	} 

  }

/**
*  External ID type
*  @return External ID type
*/
  public string Type {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnExternalId_Type_get(swigCPtr) );
	} 

  }

/**
*  External ID value
*  @return External ID value
*/
  public string Value {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnExternalId_Value_get(swigCPtr) );
	} 

  }

}

}
